SH4RE/SH4RE/CreateListing/CreateListingView.swift

```diff
@@ -271,26 +272,31 @@ struct CreateListingView: View {
                 calAvail = allDays.filter{ !calendar.isDateInWeekend($0) }
             }
         }
-        let listingFields = ["Title": title, "Description" : description, "Price" : cost, "Category" : categorySelection, "Availability": calAvail, "Address": ["postalCode": postalCode, "latitude": lat, "longitude": lon], "UID": getCurrentUserUid()] as [String : Any]
-        let documentID = documentWrite(collectionPath: "Listings", data: listingFields)
-        
-        // upload images and add paths to data fields
-        var index = 1
-        var imgPath = ""
-        var arrayImgs:[String] = []
-        for pic in pictures {
-            imgPath = "listingimages/" + documentID + "/" + String(index) + ".jpg"
-            arrayImgs.append(imgPath)
-            storageManager.upload(image: pic, path: imgPath)
-            index += 1
-        }
-        if (documentUpdate(collectionPath: "Listings", documentID: documentID, data: ["image_path" : arrayImgs])) {
-            NSLog("error");
-        }
-        showPostAlertX = true
-
-        //reset inputs
-        resetInputs()
+        getUserName(uid: getCurrentUserUid(), completion: { name in
```

Comment:
If we don't use completion handlers Firebase retrieves data asynchronously and will continue running the code before the user id name has been returned and the app will break. Pretty much any firebase read needs to be in a completion handler unless its updating a State var

Reply:
@bbryanb333 if we are just grabbing the name of the user, would it be better to store the user's name along with the other user's data that is saved onto the device rather than grabbing it every time and having to encapsulate code inside of this wrapper?

Reply:
Another way to do it and avoid wrapping code would be like this I think:
<img width="640" alt="Screenshot 2023-03-30 at 12 29 42 PM" src="https://user-images.githubusercontent.com/46950469/228903300-36b56c97-9047-4308-814b-a4f356e7c83e.png">

Reply:
I've tried awaits but they are new and don't love working with all firebase funcitons. Also you need to use Tasks and make calling functions async which i haven't been able to get to work.